server/server.js

```diff
@@ -1061,6 +1065,52 @@ exports.entryPage = "dashboard";
             }
         });
 
+        socket.on("addProxy", async (proxy, proxyID, callback) => {
+            try {
+                checkLogin(socket);
+
+                const proxyBean = await Proxy.save(proxy, proxyID, socket.userID);
+                await sendProxyList(socket);
+
+                if (proxy.applyExisting) {
+                    await restartMonitors(socket.userID);
```

Comment:
Just checking source code, may found some potential issues.

Should only restart monitors which belong to this proxy.
Even better, the proxy config of monitors should be updated without restarting.